### PR TITLE
Group samples by sequence length

### PR DESCRIPTION
**Pull request by sanagno**
Should decrease the amount of padding by about ~40% with some potential trade-off on the randomness between the sample order in epochs.

Feel free to suggest changes

### CODE REVIEW

**Comment by yk** [comment]
I like the minor shuffling. In addition to that, I'd also introduce before the minor shuffling that messages of the same length are shuffled within their length-group. i.e. all messages of length 50 are shuffled, all messages of length 51 are shuffled, etc. before the minor-shuffling step

File: model/model_training/configs/config.yaml [hunk 1]
```diff
@@ -54,10 +54,11 @@ defaults:
   samples_mixing: false # uses collator that mixes samples in the batch to create a single sample with possible multiple tasks within
   verbose: false
   output_dir: saved_model
-  use_custom_sampler: false
+  use_custom_sampler: true
```
Comment (andreaskoepf) [inline model/model_training/configs/config.yaml:57]: as long as the custom sampler is broken (e.g. doesn't use all samples if only single dataset is active) I would rather keep it disabled by default.

**Comment by sanagno** [comment]
Also solves #1991

**Review by theblackcat102 — APPROVED** [review]
Looks good to me

**Pull request merged**